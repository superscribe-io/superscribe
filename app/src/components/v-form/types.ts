import { DeepPartial, Field } from '@directus/types';

export type FormField = DeepPartial<Field> & {
	field: string;
	name: string;
	hideLabel?: boolean;
	hideLoader?: boolean;
};
