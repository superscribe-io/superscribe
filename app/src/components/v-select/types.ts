export type Option = {
	value: string | number | null;
	icon?: string;
	text?: string;
	disabled?: boolean;
	children?: Option[];
	divider?: boolean;
	selectable?: boolean;
	hidden?: boolean;
};
