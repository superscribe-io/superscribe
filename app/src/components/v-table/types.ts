export type Alignment = 'left' | 'center' | 'right';

export type HeaderRaw = {
	text: string;
	value: string;
	description?: string | null;
	align?: Alignment;
	sortable?: boolean;
	width?: number | null;
	[key: string]: any;
};

export type Header = Required<HeaderRaw>;

export type Item = {
	[key: string]: any;
};

export type ItemSelectEvent = {
	value: boolean;
	item: Item;
};

export type Sort = {
	by: string | null;
	desc: boolean;
};
