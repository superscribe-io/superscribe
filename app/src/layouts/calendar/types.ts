export type LayoutOptions = {
	template?: string;
	startDateField?: string;
	endDateField?: string;
	firstDay?: number;
	viewInfo?: {
		type: string;
		startDateStr: string;
	};
};
