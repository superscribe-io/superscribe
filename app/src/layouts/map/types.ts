import { CameraOptions } from 'maplibre-gl';

export type LayoutQuery = {
	fields: string[];
	sort: string[];
	limit: number;
	page: number;
};

export type LayoutOptions = {
	cameraOptions?: CameraOptions & { bbox: any };
	geometryField?: string;
	autoLocationFilter?: boolean;
	clusterData?: boolean;
	animateOptions?: any;
	displayTemplate?: string;
};
