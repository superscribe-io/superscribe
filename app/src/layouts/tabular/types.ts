export type LayoutOptions = {
	widths?: {
		[field: string]: number;
	};
	align?: {
		[field: string]: 'left' | 'center' | 'right';
	};
	limit?: number;
	spacing?: 'comfortable' | 'cozy' | 'compact';
};

export type LayoutQuery = {
	fields: string[];
	sort: string[];
	page: number;
	limit: number;
};
