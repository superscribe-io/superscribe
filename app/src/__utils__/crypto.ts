import crypto from 'node:crypto';

export const cryptoStub = {
	value: {
		getRandomValues: (arr: any[]) => crypto.randomBytes(arr.length),
	},
};
