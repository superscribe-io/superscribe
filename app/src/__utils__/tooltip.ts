import { Directive } from 'vue';

export const Tooltip: Directive = {
	mounted() {
		return undefined;
	},
};
