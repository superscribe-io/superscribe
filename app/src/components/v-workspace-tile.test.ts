import { test, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import VWorkspaceTile from './v-workspace-tile.vue';
import { GlobalMountOptions } from '@vue/test-utils/dist/types';
import { createI18n } from 'vue-i18n';
import { Tooltip } from '../__utils__/tooltip';

const i18n = createI18n({ legacy: false });

const props = {
	id: '1',
	x: 1,
	y: 1,
	width: 10,
	height: 10,
};

const global: GlobalMountOptions = {
	stubs: ['v-icon', 'v-menu', 'v-text-overflow', 'v-list', 'v-list-item', 'v-list-item-icon', 'v-list-item-content'],
	plugins: [i18n],
	directives: {
		Tooltip,
	},
};

test('Mount component', () => {
	expect(VWorkspaceTile).toBeTruthy();

	const wrapper = mount(VWorkspaceTile, {
		props,
		global,
	});

	expect(wrapper.html()).toMatchSnapshot();
});
