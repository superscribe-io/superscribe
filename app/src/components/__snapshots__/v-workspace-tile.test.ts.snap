// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Mount component 1`] = `
"<div data-v-a110deb5=\\"\\" class=\\"v-workspace-tile draggable br-tl br-tr br-br br-bl\\" style=\\"--pos-x: 1; --pos-y: 1; --width: 10; --height: 10;\\" data-move=\\"\\">
  <div data-v-a110deb5=\\"\\" class=\\"header\\">
    <v-icon-stub data-v-a110deb5=\\"\\" class=\\"icon\\" style=\\"--v-icon-color: var(--primary);\\" name=\\"space_dashboard\\" small=\\"\\"></v-icon-stub>
    <v-text-overflow-stub data-v-a110deb5=\\"\\" class=\\"name\\" text=\\"\\"></v-text-overflow-stub>
    <div data-v-a110deb5=\\"\\" class=\\"spacer\\"></div>
    <!--v-if-->
  </div>
  <!--v-if-->
  <div data-v-a110deb5=\\"\\" class=\\"resize-details\\"> (0:0) 10Ã—10</div>
  <!--v-if-->
  <div data-v-a110deb5=\\"\\" class=\\"tile-content has-header\\">
    <!--v-if-->
  </div>
</div>"
`;
